src/container/cli.c

```diff
@@ -816,15 +817,17 @@ cont_open_complete(tse_task_t *task, void *data)
 		arg->coa_info->ci_md_mtime = out->coo_md_mtime;
 	}
 
-	/* log metadata times. NB: ctime_r inserts \n so don't add it in the D_DEBUG format */
 	rc = daos_hlc2timestamp(arg->coa_info->ci_md_otime, &otime_sec);
 	if (rc)
 		goto out;
 	rc = daos_hlc2timestamp(arg->coa_info->ci_md_mtime, &mtime_sec);
 	if (rc)
 		goto out;
-	D_ASSERT(otime_str == ctime_r((const time_t *)&otime_sec, otime_str));
-	D_ASSERT(mtime_str == ctime_r((const time_t *)&mtime_sec, mtime_str));
+	out_str = ctime_r((const time_t *)&otime_sec, otime_str);
```

Comment:
why is this change required?

Reply:
Without it otime_str is undefined and you get valgrind errors when it's logged on the D_DEBUG below.  This is the case where only debug output is affected by the side-effect of code within the assert.